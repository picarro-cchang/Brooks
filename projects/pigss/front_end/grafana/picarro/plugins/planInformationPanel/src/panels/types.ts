import { any } from "prop-types";

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}
