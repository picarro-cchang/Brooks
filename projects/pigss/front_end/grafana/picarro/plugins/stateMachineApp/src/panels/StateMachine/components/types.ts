export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
