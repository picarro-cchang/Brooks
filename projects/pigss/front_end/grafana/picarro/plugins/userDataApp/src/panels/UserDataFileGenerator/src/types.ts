import { GrafanaTheme } from "@grafana/ui";
import { TimeRange } from "@grafana/data";

export interface DataGeneratorPanelProps {
  timeRange: TimeRange;
}

export interface DataGeneratorLayoutProps {
  theme: GrafanaTheme;
  options: DataGeneratorPanelProps;
}
