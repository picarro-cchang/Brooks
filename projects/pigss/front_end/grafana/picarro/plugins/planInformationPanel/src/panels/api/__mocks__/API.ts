import data from "./mockData.json";
import plan from "./mockPlan.json";

const apiLoc = `http://localhost:8000/controller`;

const PicarroAPI = {
  getRequest(url: string) {
    switch (url) {
      case apiLoc + "/uistatus": {
        return Promise.resolve(new Response(JSON.stringify(data)));
      }
      case apiLoc + "/plan": {
        return Promise.resolve(new Response(JSON.stringify(plan)));
      }
    }
  }
};

export default PicarroAPI;
