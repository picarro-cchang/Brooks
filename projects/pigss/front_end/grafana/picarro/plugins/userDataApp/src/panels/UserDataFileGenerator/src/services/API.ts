import { notifyError } from '../utils/Notifications';

export const API = {
  get(url: string) {
    return fetch(url, { method: 'GET' })
      .then(response => {
        if (!response.ok) {
          throw Error('Netwoek GET request failed.');
        }
        return response;
      })
      .catch(err => {
        notifyError(err.message);
        throw Error('Netwoek GET request failed.');
      });
  },
};
