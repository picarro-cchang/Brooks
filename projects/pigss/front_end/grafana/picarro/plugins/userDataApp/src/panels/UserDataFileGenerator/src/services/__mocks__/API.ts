import {URL} from '../../constants'

export const API = {
    get(url: string) {
      switch(url) {
        case URL.GET_SAVED_FILES: {
            const prom = Promise.resolve(Object({"files": ["pigss-ms-02-05-2020_121737-02-05-2020_131737.csv", "pigss-ms-02-05-2020_121814-02-05-2020_131814.csv"]}))
            return prom
        }
        case URL.GET_FIELD_KEYS: {
          const promise = new Promise((resolve) => {
            resolve({"keys": ["CavityPressure", "WarmBoxTemp", "HCl", "H2O", "SO2", "CH4"]})
          });
          return promise
        }
        case URL.GET_ANALYZERS: {
          return new Promise((resolve) => {
            resolve({"analyzers": ["AMADS3001", "AMSADS3003", "BFADS3003", "SBDS3002", "UADS3003"]});
          });
        }
        case URL.GET_FILE + '?name=': {
          return new Promise((resolve) => {
          resolve([["time","analyzer","valve_pos","WarmBoxTemp","CavityTemp"], 
            ["1580936855480","AMSADS3003","18","45","80"],
            ["1580936854180","AMSADS3003","18","45","80"],
            ["1580936852880","AMSADS3003","18","45","80"]])
          });
        }
        case URL.GENERATE_FILE + '?keys=CavityTemp&analyzer=AMSADS3003&port=2&from=&to=': {
          return new Promise((resolve) => {resolve({"filename": "pigss-ms-02-13-2020_061135-02-13-2020_121135.csv"})});
        }
      }
    },
    post(url: string, data: object={}) {
      return new Promise((resolve) => resolve([{"text": "2: Bank 1 Ch. 2", "value": "2"}, {"text": "4: Bank 1 Ch. 4", "value": "4"}, {"text": "6: Bank 1 Ch. 6", "value": "6"}, {"text": "8: Bank 1 Ch. 8", "value": "8"}, {"text": "18: Bank 3 Ch. 2", "value": "18"}, {"text": "20: Bank 3 Ch. 4", "value": "20"}, {"text": "22: Bank 3 Ch. 6", "value": "22"}, {"text": "24: Bank 3 Ch. 8", "value": "24"}, {"text": "26: Bank 4 Ch. 2", "value": "26"}, {"text": "28: Bank 4 Ch. 4", "value": "28"}, {"text": "30: Bank 4 Ch. 6", "value": "30"}, {"text": "32: Bank 4 Ch. 8", "value": "32"}]))
    },
  };
  