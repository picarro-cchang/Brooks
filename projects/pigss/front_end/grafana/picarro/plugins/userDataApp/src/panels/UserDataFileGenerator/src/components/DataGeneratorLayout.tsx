import React, { PureComponent, Fragment } from 'react';
import { TimeRange, TIME_FORMAT, dateTime, dateMath } from '@grafana/data';
import { TimePicker, FormLabel, PanelOptionsGroup, Select, Button } from '@grafana/ui';

import { notifyError, notifySuccess } from '../utils/Notifications';
import { DataGeneratorLayoutProps } from '../types';
import { DEFAULT_TIME_OPTIONS } from '../constants';
import { DataGeneratorService } from '../services/DataGeneratorService';

import './Layout.css';

interface Props extends DataGeneratorLayoutProps { }

const labelWidth_6 = 6;
const labelWidth_7 = 7;
const selectWidth = 12;

export default class DataGeneratorLayout extends PureComponent<Props, any> {
  constructor(props: Props) {
    super(props);
    this.state = {
      timeRange: this.props.options.timeRange,
      keys: [],
      keyOptions: [],
      files: [],
    };
  }

  // Code to download the file
  downloadData = (blob: any, fileName: string) => {
    const a = document.createElement('a');

    document.getElementById('file-list').appendChild(a);
    // @ts-ignore
    a.style = 'display: none';
    const url = window.URL.createObjectURL(blob);
    a.href = url;
    a.download = fileName;
    a.click();
    notifySuccess(`File successfully downloaded in Downloads folder.`);
    window.URL.revokeObjectURL(url);
  };

  generateFile = () => {
    const { timeRange, keys } = this.state;
    const {from, to } = {
      from: dateMath.parse(timeRange.raw.from),
      to: dateMath.parse(timeRange.raw.to),
      raw: timeRange.raw
    } as TimeRange;
    
    // @ts-ignore
    let fromTime = from._d.getTime();
    // @ts-ignore
    let toTime = to._d.getTime();

    if (fromTime === toTime || keys.length === 0) {
      notifyError('Invalid Query parameters');
      return;
    }

    const queryParams = {
      from: fromTime * 1000000,
      to: toTime * 1000000,
      keys: keys,
    };

    // Call generate file api
    DataGeneratorService.generateFile(queryParams).then((response: any) => {
      response.json().then((data: any) => {
        // call getFile to download the data
        if (data.filename !== undefined) {
          this.getFileNames();
          this.getFile(data.filename);
        } else {
          if (data !== undefined && data.hasOwnProperty('message')) {
            notifyError(data.message);
          }
        }
      });
    });
  };

  getFile = (fileName: string) => {
    DataGeneratorService.getFile(fileName).then((response: any) => {
      response.blob().then((data: any) => {
        this.downloadData(data, fileName);
      });
    });
  };

  getFileNames = () => {
    // Get file names
    DataGeneratorService.getSavedFiles().then((response: any) => {
      response.json().then((files: any) => {
        this.setState(() => {
          return files;
        });
      });
    });
  };

  onDateChange = (timeRange: TimeRange) => {
    this.setState({ timeRange });
  };

  onKeysChange = (keys: any) => {
    this.setState(() => {
      return { keys };
    });
  };

  componentWillMount() {
    // Get file names
    this.getFileNames();

    // Get Key Options
    DataGeneratorService.getKeys().then((response: any) => {
      response.json().then((data: any) => {
        const keyOptions = data.keys.map((x: string) => {
          return {
            value: x,
            label: x,
          };
        });
        this.setState({ keyOptions: keyOptions });
      });
    });
  }

  render() {
    const { files, timeRange, keyOptions } = this.state;

    const styleObj = {
      overflow: 'scroll !important',
      height: 'inherit',
    };

    if (typeof timeRange.from === 'string') {
      timeRange.from = dateTime(timeRange.from);
    }
    if (typeof timeRange.to === 'string') {
      timeRange.to = dateTime(timeRange.to);
    }

    const fileItems = files.map((fn: string, i: number) => {
      return (
        <li key={fn} className="file-item" value={fn} onClick={() => this.getFile(fn)}>
          {fn}
        </li>
      );
    });

    return (
      <Fragment>
        <h3 className="text-center">User Data File Generator</h3>
        <PanelOptionsGroup title="Generate New File">
          <div className="row">
            <div className="gf-form col-md-3 col-sm-12">
              <FormLabel width={labelWidth_6}>Species</FormLabel>
              <Select
                width={selectWidth}
                options={keyOptions}
                onChange={this.onKeysChange}
                value={keyOptions.find((option: any) => option.value === 'key')}
                isMulti={true}
                backspaceRemovesValue
              />
            </div>
            <div className="gf-form col-md-6 col-sm-12 time-picker-container">
              <FormLabel width={labelWidth_7}>Time Range</FormLabel>
              <TimePicker
                timeZone="browser"
                selectOptions={DEFAULT_TIME_OPTIONS}
                onChange={this.onDateChange}
                value={this.state.timeRange}
                onMoveBackward={() => console.log('Move Backward')}
                onMoveForward={() => console.log('Move forward')}
                onZoom={() => console.log('Zoom')}
              />
            </div>

            <div className="gf-form col-md-3 col-sm-12">
              <Button size="md" variant="primary" value="Generate" onClick={this.generateFile} disabled={!this.state.keys.length}>
                Generate
              </Button>
            </div>
          </div>
        </PanelOptionsGroup>
        <PanelOptionsGroup title="Recently Generated Files">
          <div className="row" style={styleObj}>
            <div className="gf-form col-md-12 col-sm-12">
              <ul id="file-list" style={{ listStyle: 'none' }}>
                {fileItems}
              </ul>
            </div>
          </div>
        </PanelOptionsGroup>
      </Fragment>
    );
  }
}
