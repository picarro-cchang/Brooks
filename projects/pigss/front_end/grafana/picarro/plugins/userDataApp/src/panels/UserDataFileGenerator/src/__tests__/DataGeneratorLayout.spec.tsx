import React from 'react';
import DataGeneratorLayout from '../components/DataGeneratorLayout';
import { shallow, mount } from 'enzyme';
import { DataGeneratorLayoutProps } from '../types';
import { TimeRange, dateTime, dateMath, TimeFragment } from '@grafana/data';
import 'jest-styled-components';
import "jest-fetch-mock";
import { API } from '../services/API';

jest.mock('./../services/DataGeneratorService.ts');

const defaultProps: DataGeneratorLayoutProps = {
    theme: null,
    options: {
        timeRange: {
            from: dateTime("2020-02-11T14:31:13.973Z"),
            to: dateTime("2020-02-11T20:31:13.973Z"),
            raw: { from: 'now-6h' as TimeFragment, to: 'now' as TimeFragment },
        }
    }
};

describe('<DataGeneratorLayout />', () => {
  const wrapper = shallow(<DataGeneratorLayout {...defaultProps}/>);
  const instance = wrapper.instance() as DataGeneratorLayout;
  const getFileNames = jest.spyOn(instance, 'getFileNames');

  it('Renders correctly', () => {
      return expect(wrapper).toMatchSnapshot();
  });

  it('generateFile', () => {
      const generateFile = jest.spyOn(instance, 'generateFile');
      instance.forceUpdate();
      instance.generateFile()      
      expect(generateFile).toHaveBeenCalled();
  });
  
  it('getFileNames', async () => {
      await instance.getFileNames();
      expect(getFileNames).toHaveBeenCalled();
      await expect(instance.state.files).toEqual('files')
  });
});

