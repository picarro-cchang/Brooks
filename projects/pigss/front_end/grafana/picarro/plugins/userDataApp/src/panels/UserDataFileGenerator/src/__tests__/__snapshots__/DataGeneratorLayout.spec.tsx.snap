// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataGeneratorLayout /> Renders correctly 1`] = `
<Fragment>
  <h3
    className="text-center"
  >
    Export Data
  </h3>
  <PanelOptionsGroup
    title="Generate New File"
  >
    <PanelOptionsGroup>
      <div
        className="gf-form"
      >
        <FormLabel
          width={6}
        >
          Species
        </FormLabel>
        <StateManager
          backspaceRemovesValue={true}
          components={
            Object {
              "Option": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isMulti={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "All",
                "value": "All",
              },
              Object {
                "label": "CavityPressure",
                "value": "CavityPressure",
              },
              Object {
                "label": "WarmBoxTemp",
                "value": "WarmBoxTemp",
              },
              Object {
                "label": "HCl",
                "value": "HCl",
              },
              Object {
                "label": "H2O",
                "value": "H2O",
              },
              Object {
                "label": "SO2",
                "value": "SO2",
              },
              Object {
                "label": "CH4",
                "value": "CH4",
              },
            ]
          }
          styles={
            Object {
              "control": [Function],
              "menu": [Function],
              "option": [Function],
            }
          }
        />
      </div>
    </PanelOptionsGroup>
    <PanelOptionsGroup>
      <div
        className="gf-form"
      >
        <FormLabel
          width={6}
        >
          Analyzer
        </FormLabel>
        <StateManager
          backspaceRemovesValue={true}
          components={
            Object {
              "Option": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isMulti={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "All",
                "value": "All",
              },
              Object {
                "label": "SI2306",
                "value": "SI2306",
              },
              Object {
                "label": "SI2108",
                "value": "SI2108",
              },
              Object {
                "label": "SI3401",
                "value": "SI3401",
              },
              Object {
                "label": "BFADS",
                "value": "BFADS",
              },
              Object {
                "label": "SI5450",
                "value": "SI5450",
              },
            ]
          }
        />
      </div>
    </PanelOptionsGroup>
    <PanelOptionsGroup>
      <div
        className="gf-form"
      >
        <FormLabel
          width={6}
        >
          Port
        </FormLabel>
        <StateManager
          backspaceRemovesValue={true}
          components={
            Object {
              "Option": [Function],
            }
          }
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          isMulti={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "All",
                "value": "All",
              },
              Object {
                "label": "2: Bank 1 Ch. 2",
                "value": "2",
              },
              Object {
                "label": "4: Bank 1 Ch. 4",
                "value": "4",
              },
              Object {
                "label": "6: Bank 1 Ch. 6",
                "value": "6",
              },
              Object {
                "label": "8: Bank 1 Ch. 8",
                "value": "8",
              },
              Object {
                "label": "18: Bank 3 Ch. 2",
                "value": "18",
              },
              Object {
                "label": "20: Bank 3 Ch. 4",
                "value": "20",
              },
              Object {
                "label": "22: Bank 3 Ch. 6",
                "value": "22",
              },
              Object {
                "label": "24: Bank 3 Ch. 8",
                "value": "24",
              },
              Object {
                "label": "26: Bank 4 Ch. 2",
                "value": "26",
              },
              Object {
                "label": "28: Bank 4 Ch. 4",
                "value": "28",
              },
              Object {
                "label": "30: Bank 4 Ch. 6",
                "value": "30",
              },
              Object {
                "label": "32: Bank 4 Ch. 8",
                "value": "32",
              },
            ]
          }
        />
      </div>
    </PanelOptionsGroup>
    <PanelOptionsGroup>
      <div
        className="gf-form"
      >
        <FormLabel
          tooltip="Toggle the switch to download raw data"
          width={10}
        >
          Processed Data
        </FormLabel>
        <Switch
          checked={true}
          label=""
          onChange={[Function]}
        />
      </div>
    </PanelOptionsGroup>
    <PanelOptionsGroup>
      <div
        className="time-generate-btn"
      >
        <div
          className="gf-form time-picker-container"
        >
          <FormLabel
            width={7}
          >
            Time Range
          </FormLabel>
          <TimePicker
            onChange={[Function]}
            onMoveBackward={[Function]}
            onMoveForward={[Function]}
            onZoom={[Function]}
            selectOptions={
              Array [
                Object {
                  "display": "Last 5 minutes",
                  "from": "now-5m",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 15 minutes",
                  "from": "now-15m",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 30 minutes",
                  "from": "now-30m",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 1 hour",
                  "from": "now-1h",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 3 hours",
                  "from": "now-3h",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 6 hours",
                  "from": "now-6h",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 12 hours",
                  "from": "now-12h",
                  "section": 3,
                  "to": "now",
                },
                Object {
                  "display": "Last 24 hours",
                  "from": "now-24h",
                  "section": 3,
                  "to": "now",
                },
              ]
            }
            timeZone="browser"
            value={
              Object {
                "from": "2020-02-11T14:31:13.973Z",
                "raw": Object {
                  "from": "now-6h",
                  "to": "now",
                },
                "to": "2020-02-11T20:31:13.973Z",
              }
            }
          />
          <div
            className="gf-form col-md-1"
          >
            <Button
              disabled={true}
              onClick={[MockFunction]}
              size="md"
              value="Generate"
              variant="primary"
            >
              Generate
            </Button>
          </div>
        </div>
      </div>
    </PanelOptionsGroup>
  </PanelOptionsGroup>
  <PanelOptionsGroup
    title="Recently Generated Files"
  >
    <div
      className="row"
      style={
        Object {
          "height": "inherit",
          "overflow": "scroll !important",
        }
      }
    >
      <div
        className="gf-form col-md-12 col-sm-12"
      >
        <ul
          id="file-list"
          style={
            Object {
              "listStyle": "none",
            }
          }
        >
          <li
            className="file-item"
            key="pigss-ms-02-05-2020_121737-02-05-2020_131737.csv"
            onClick={[Function]}
            value="pigss-ms-02-05-2020_121737-02-05-2020_131737.csv"
          >
            pigss-ms-02-05-2020_121737-02-05-2020_131737.csv
          </li>
          <li
            className="file-item"
            key="pigss-ms-02-05-2020_121814-02-05-2020_131814.csv"
            onClick={[Function]}
            value="pigss-ms-02-05-2020_121814-02-05-2020_131814.csv"
          >
            pigss-ms-02-05-2020_121814-02-05-2020_131814.csv
          </li>
        </ul>
      </div>
    </div>
  </PanelOptionsGroup>
</Fragment>
`;

exports[`<DataGeneratorLayout /> mount with String TimeRange 1`] = `
DataGeneratorLayout {
  "context": Object {},
  "downloadData": [Function],
  "generateFile": [Function],
  "getFile": [Function],
  "getFileNames": [Function],
  "onAnalyzersChange": [Function],
  "onDateChange": [Function],
  "onKeysChange": [Function],
  "onPortsChange": [Function],
  "onProcessedDataSwitchChange": [Function],
  "props": Object {
    "options": Object {
      "timeRange": Object {
        "from": "2020-02-11T14:31:13.973Z",
        "raw": Object {
          "from": "now-6h",
          "to": "now",
        },
        "to": "2020-02-11T20:31:13.973Z",
      },
    },
    "theme": null,
    "timeRange": Object {
      "from": "2020-02-11T14:31:13.973Z",
      "raw": Object {
        "from": "now-6h",
        "to": "now",
      },
      "to": "2020-02-11T20:31:13.973Z",
    },
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "analyzerOptions": Array [
      Object {
        "label": "All",
        "value": "All",
      },
    ],
    "analyzers": Array [],
    "files": Array [],
    "from": "2020-02-11T14:31:13.973Z",
    "isProcessedData": true,
    "keyOptions": Array [
      Object {
        "label": "All",
        "value": "All",
      },
    ],
    "keys": Array [],
    "portOptions": Array [
      Object {
        "label": "All",
        "value": "All",
      },
    ],
    "ports": Array [],
    "raw": Object {
      "from": "now-6h",
      "to": "now",
    },
    "timeRange": Object {
      "from": "2020-02-11T14:31:13.973Z",
      "raw": Object {
        "from": "now-6h",
        "to": "now",
      },
      "to": "2020-02-11T20:31:13.973Z",
    },
    "to": "2020-02-11T20:31:13.973Z",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <DataGeneratorLayout
        options={
          Object {
            "timeRange": Object {
              "from": "2020-02-11T14:31:13.973Z",
              "raw": Object {
                "from": "now-6h",
                "to": "now",
              },
              "to": "2020-02-11T20:31:13.973Z",
            },
          }
        }
        theme={null}
        timeRange={
          Object {
            "from": "2020-02-11T14:31:13.973Z",
            "raw": Object {
              "from": "now-6h",
              "to": "now",
            },
            "to": "2020-02-11T20:31:13.973Z",
          }
        }
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <React.Fragment>
        <h3
          className="text-center"
        >
          Export Data
        </h3>
        <PanelOptionsGroup
          title="Generate New File"
        >
          <PanelOptionsGroup>
            <div
              className="gf-form"
            >
              <FormLabel
                width={6}
              >
                Species
              </FormLabel>
              <StateManager
                backspaceRemovesValue={true}
                components={
                  Object {
                    "Option": [Function],
                  }
                }
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                isMulti={true}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "All",
                      "value": "All",
                    },
                  ]
                }
                styles={
                  Object {
                    "control": [Function],
                    "menu": [Function],
                    "option": [Function],
                  }
                }
              />
            </div>
          </PanelOptionsGroup>
          <PanelOptionsGroup>
            <div
              className="gf-form"
            >
              <FormLabel
                width={6}
              >
                Analyzer
              </FormLabel>
              <StateManager
                backspaceRemovesValue={true}
                components={
                  Object {
                    "Option": [Function],
                  }
                }
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                isMulti={true}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "All",
                      "value": "All",
                    },
                  ]
                }
              />
            </div>
          </PanelOptionsGroup>
          <PanelOptionsGroup>
            <div
              className="gf-form"
            >
              <FormLabel
                width={6}
              >
                Port
              </FormLabel>
              <StateManager
                backspaceRemovesValue={true}
                components={
                  Object {
                    "Option": [Function],
                  }
                }
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                isMulti={true}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "All",
                      "value": "All",
                    },
                  ]
                }
              />
            </div>
          </PanelOptionsGroup>
          <PanelOptionsGroup>
            <div
              className="gf-form"
            >
              <FormLabel
                tooltip="Toggle the switch to download raw data"
                width={10}
              >
                Processed Data
              </FormLabel>
              <Switch
                checked={true}
                label=""
                onChange={[Function]}
              />
            </div>
          </PanelOptionsGroup>
          <PanelOptionsGroup>
            <div
              className="time-generate-btn"
            >
              <div
                className="gf-form time-picker-container"
              >
                <FormLabel
                  width={7}
                >
                  Time Range
                </FormLabel>
                <TimePicker
                  onChange={[Function]}
                  onMoveBackward={[Function]}
                  onMoveForward={[Function]}
                  onZoom={[Function]}
                  selectOptions={
                    Array [
                      Object {
                        "display": "Last 5 minutes",
                        "from": "now-5m",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 15 minutes",
                        "from": "now-15m",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 30 minutes",
                        "from": "now-30m",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 1 hour",
                        "from": "now-1h",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 3 hours",
                        "from": "now-3h",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 6 hours",
                        "from": "now-6h",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 12 hours",
                        "from": "now-12h",
                        "section": 3,
                        "to": "now",
                      },
                      Object {
                        "display": "Last 24 hours",
                        "from": "now-24h",
                        "section": 3,
                        "to": "now",
                      },
                    ]
                  }
                  timeZone="browser"
                  value={
                    Object {
                      "from": "2020-02-11T14:31:13.973Z",
                      "raw": Object {
                        "from": "now-6h",
                        "to": "now",
                      },
                      "to": "2020-02-11T20:31:13.973Z",
                    }
                  }
                />
                <div
                  className="gf-form col-md-1"
                >
                  <Button
                    disabled={true}
                    onClick={[Function]}
                    size="md"
                    value="Generate"
                    variant="primary"
                  >
                    Generate
                  </Button>
                </div>
              </div>
            </div>
          </PanelOptionsGroup>
        </PanelOptionsGroup>
        <PanelOptionsGroup
          title="Recently Generated Files"
        >
          <div
            className="row"
            style={
              Object {
                "height": "inherit",
                "overflow": "scroll !important",
              }
            }
          >
            <div
              className="gf-form col-md-12 col-sm-12"
            >
              <ul
                id="file-list"
                style={
                  Object {
                    "listStyle": "none",
                  }
                }
              />
            </div>
          </div>
        </PanelOptionsGroup>
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
